libminifi/include/utils/GeneralUtils.h

```diff
@@ -62,6 +62,15 @@ struct identity {
 };
 #endif /* < C++20 */
 
+#if __cplusplus > 201703L
+using std::type_identity;
+#else
+template<typename T>
+struct type_identity {
+  using type = T;
+};
+#endif /* < C++20 */
```

Comment:
There is a [feature test macro](https://en.cppreference.com/w/cpp/feature_test) for this: `__cpp_lib_type_identity`
```suggestion
#if __cpp_lib_type_identity >= 201806L
using std::type_identity;
#else
template<typename T>
struct type_identity {
  using type = T;
};
#endif /* has std::type_identity */
```
It's not used elsewhere because I believe feature test macros were standardized in C++20, so I see no point in checking for features that were introduced earlier. And I don't know the macro for std::identity above. (Maybe ranges?)